legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/to/PureGrammarComposer.java

```diff
@@ -105,6 +107,55 @@ public String renderPureModelContextData(PureModelContextData pureModelContextDa
         return composedSections.select(section -> !section.isEmpty()).makeString("\n\n");
     }
 
+    public Map<String, String> renderPureModelContextDataForEachElement(PureModelContextData pureModelContextData)
+    {
+        List<PackageableElement> elements = pureModelContextData.getElements();
+        Set<PackageableElement> elementsToCompose = new HashSet<>(elements);
+        Map<String, String> composedSections = Maps.mutable.empty();
+        if (ListIterate.anySatisfy(elements, e -> e instanceof SectionIndex))
+        {
+            Map<String, PackageableElement> elementByPath = new HashMap<>();
```

Comment:
wait, what's all this complexity above, you take the PMCD and you produce a `Map<String, String>`, section index is completely irrelevant here in terms of ordering, only good in terms of getting the `import` statements right? I expect the logic to be utterly simple here, as if `pureModelContextData.elements.forEach`, am I missing something? 🤷

Reply:
yeah that makes more sense